### ISSUE TITLE
Move org.eclipse.jdt.doc.user to JDT repository

### ISSUE THREAD

**Issue by laeubi** [eclipse-platform/eclipse.platform.releng.aggregator#1487]
Looking at https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/blob/master/eclipse.platform.common/bundles/org.eclipse.jdt.doc.user/pom.xml this seems to require nothing special from aggregator build itself.

So it seems suitable to be maintained at the jdt project itself.

WDYT @akurtakov @iloveeclipse 
@mickaelistria  can you probabbyl help with creating a PR with full history if we see if/how this can work out at the JDT side? We then probably need to update some references in aggregator afterwards.

**Comment by mickaelistria** [comment]
jdt.doc.isv as well.
But I see content in parent pom at https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/blob/master/eclipse.platform.common/pom.xml which seems to be common to all docs bundles. Do we want to un-factorize this part? (I personally don't have a strong opinion here).

**Comment by laeubi** [comment]
> jdt.doc.isv as well.

That is not *that* easy but I'm working on it, `org.eclipse.jdt.doc.user` on the other hand seems like a low haning fruit if I do not overlook anything?

**Comment by mickaelistria** [comment]
IMO, it's better to migrate do both or none here. Migrating just one would make things a bit inconsistent and I don't think it would actually help overall.

**Comment by laeubi** [comment]
> Migrating just one would make things a bit inconsistent and I don't think it would actually help overall.

Why? as shown `org.eclipse.jdt.doc.user` seem not to relate to anything else (aka self-contained), while `jdt.doc.isv` has dependencies to literally everything in the aggregator (see [org.eclipse.jdt.doc.isv/cbi_basedirs.properties](https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/blob/master/eclipse.platform.common/bundles/org.eclipse.jdt.doc.isv/cbi_basedirs.properties)).

> But I see content in parent pom at https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/blob/master/eclipse.platform.common/pom.xml which seems to be common to all docs bundles. Do we want to un-factorize this part? 

I think we disabled indexing altogether already so this is just dead code.

**Comment by mickaelistria** [comment]
> Why? 

Imagine some contributor saying "I want to contribute to JDT doc". The current state is to just go to the right repo and edit it. If we split, then the same "I want to contribute to JDT doc" now brings additional questions before finding the right thing to do. So I don't think it makes things simpler is one more step is necessary before being able to produce value.

**Comment by laeubi** [comment]
> The current state is to just go to the right repo and edit it. 

You seem to asume that the user already know what is "the right repo", currently **always** the user needs to head over to a completeley different alien place, while with moving the user doc at laest that can be found already next to the code.

But of course this was jsut an idea of something we can act on **now** than sometime in the future (if ever).

**Comment by mickaelistria** [comment]
> You seem to asume that the user already know what is "the right repo"

No. I also think about reviewers (us) who, if a user wants to contribute to the doc, will have to constantly ask them "What do you want to contribute? Is this final user documentation or API/framework documentation?" and start a longer conversation while the current state is just to answer 1 link.
Moreover, some changes might affect both user and isv at the same time and one might prefer 1 PR for both rather than 2 PRs in 2 distinct repos.
That's why I think that both should move when they're ready; but moving only user doc now seems like a good intention, but with no so positive outcome.

**Comment by vogella** [comment]
Should the PDE docs also be moved to the PDE repo?

**Comment by laeubi** [comment]
@vogella see
- https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/issues/1497
